### PR TITLE
VideoPress: Fix dashboard fatal mistake when reading malformed local video

### LINKED ISSUE #28707: VideoPress: Improve error handling when reading local media library files
A user had a problem upon installing VideoPress where the dashboard would return a `500` error.
It turns out to be related to process of reading the local library files, where bogus files cause an error to be thrown:
https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/class-uploader.php#L77-L79

Instead of throwing an exception, we need to mark the bad file and ignore it.

ref https://github.com/Automattic/greenhouse/issues/1563#issuecomment-1412633164

### PR DESCRIPTION
**Pull request by dhasilva**
Fixes #28707

## Proposed changes:

* Checks if `$media_details` or `$jetpack_videopress` are objects instead of arrays, marking the file as malformed and converting the properties to array, avoiding a `Cannot use object of type stdClass as array` exception.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1676573126334559-slack-C03TA48NSUX

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Testing locally:
* Upload 2 videos to the local library (Media > Add New)

For the first video we will simulate a database problem:
* Get the `post_id` of the video
* Update the database entry in the `wp_postmeta` table for the video's `post_id`, setting `_wp_attachment_metadata` to an empty string (`localhost:8181/` locally)
* 

For the second video we will remove the file to cause a read error:
* Go to `/wp-content/uploads/<year>/<month>`. Locally and today that would be `tools/docker/wordpress/wp-content/uploads/2023/02`
* Delete or rename the video file: `sudo mv file.mp4 file.mp4.bak`

* Go to the VideoPress dashboard
* Check that there is no fatal error and the dashboard is accessible
* Check that both videos are listed with a warning and cannot be uploaded

Before | After
-|-
 |

### CODE REVIEW

File: projects/packages/videopress/src/class-data.php [hunk 1]
```diff
@@ -237,15 +237,23 @@ public static function get_initial_state() {
 		// Tweak local videos data.
 		$local_videos = array_map(
 			function ( $video ) {
+				$video              = (array) $video;
 				$id                 = $video['id'];
 				$media_details      = $video['media_details'];
 				$jetpack_videopress = $video['jetpack_videopress'];
```
Comment (retrofox) [inline projects/packages/videopress/src/class-data.php:243]: I wonder if we should cast straight here:

```suggestion
				$media_details      = (array) $video['media_details'];
				$jetpack_videopress = (array) $video['jetpack_videopress'];
```

Comment (dhasilva) [reply projects/packages/videopress/src/class-data.php:243]: I used the `if` to add the `$read_error`. If we cast straight here it is not possible to know if there is a problem.

Comment (retrofox) [reply projects/packages/videopress/src/class-data.php:243]: fair enough

**Review by retrofox — APPROVED** [review]
Tested and works as described. Probably we'd like to get the @lhkowalski feedback, too.
Thanks for taking over this one 🤘

**Review by lhkowalski — APPROVED** [review]

**Pull request merged**